### PR TITLE
Use productized bits and images

### LINKED ISSUE #62: Intersmash should run with products and not just with community projects
**Is your feature request related to a problem? Please describe.**
Intersmah should be able to run with products' bits and deliverables that originate from the currently supported community projects (e.g.: WildFLy -> EAP, Keycloak -> RHSSO etc.)

**Describe the solution you'd like**
When using Intersmash for provisioning an interoperability testing scenario, users should be able to use products' bits and deliverables (e.g.: images) for configuring _what_ actually would be provisioned.
This means that the provisioning components should be validated with such configured values.
This should be done by adding/modifying tests to the testsuite module, as it has been done in #21.
BTW - this affects the deployments module, which is used by some provisioners, namely the WildFly one. Such deployments are currently built with community artifacts, while they should use productized ones when the tests involve products rather than community projects. This could be done by defining a dedicated Maven profile to cover the differences in configuration.
Given the provisioners will try to use images which are available from a non-public registry, the Intersmash OpenShift CI infra configuration will be modified as needed, e.g.: for storing the secrets to access such registries.

**Describe alternatives you've considered**
None, this is a required feature for being able and use Intersmash for provisioning and test scenarios with products rather than community projects.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by fabiobrz**
## Description
Modifying the OpenShift CI automation in order to use a pull secret for pulling _released_ images from secured registries.
Updating the WildFly/EAP8 images related properties to experiment with EAP 8 Beta images in registry.redhat.io.

 Fix #62

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] Pull Request contains a description of the changes
 - [ ] Pull Request does not include fixes for multiple issues/topics
 - [x] Code is self-descriptive and/or documented
 - [ ] I have implemented unit tests to cover my changes
 - [x] I tested my code in OpenShift

### CODE REVIEW

**Comment by fabiobrz** [comment]
Pulling _released_ product images from a secured registry works, see https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/Intersmash_intersmash/70/pull-ci-Intersmash-intersmash-main-intersmash-e2e/1684150504336134144 :heavy_check_mark: 

**Next steps**

- [ ] make the deployments build configurable so that productized bits can be used instead of community ones
- [ ] isolate the productized variant of the e2e test process into a new job 
- [ ] evaluate whether to implement support for testable images and bits (more secured registries, potentially new credentials for accessing a different Maven repo?) right away or to postpone it.

**Comment by fabiobrz** [comment]
Hi @tommaso-borgato - this is ready to be reviewed, see my summary at: https://github.com/Intersmash/intersmash/pull/70#issuecomment-1651565391

TL;DR:

* https://github.com/Intersmash/intersmash/commit/0da50dbece420d03bf06b65e58edbcad35e00775 and https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/Intersmash_intersmash/70/pull-ci-Intersmash-intersmash-main-intersmash-e2e/1696837227113353216 proves that released _productized_ bits and deliverables are supported

* https://github.com/Intersmash/intersmash/commit/53d8a5534f21b1c0d6f9e649fe343ad01c8156da and https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/Intersmash_intersmash/70/pull-ci-Intersmash-intersmash-main-intersmash-e2e/1696806954539683840 proves that released _community_ bits and deliverables are supported.

We can remove those two `[TEMP]` commits that I've used fr verification right before merging.

**Comment by tommaso-borgato** [comment]
| Java Property | XTF Property | Community | Product | Notes |
|--------|--------|--------|--------|
| * ACTIVEMQ_IMAGE_URL | intersmash.activemq.image | | registry.redhat.io/amq7/amq-broker-rhel8:7.11.0 | |
| * ACTIVEMQ_INIT_IMAGE_URL | intersmash.activemq.init.image | | registry.redhat.io/amq7/amq-broker-init-rhel8:7.11.0 | |
| ** ACTIVEMQ_OPERATOR_CATALOG_SOURCE_NAME | intersmash.activemq.operators.catalog_source | intersmash-activemq-operator-index | redhat-operators | |
| ** ACTIVEMQ_OPERATOR_CHANNEL | intersmash.activemq.operators.channel | upstream | 7.11.x | |
| *** ACTIVEMQ_OPERATOR_INDEX_IMAGE | intersmash.activemq.operators.index_image | quay.io/jbossqe-eap/intersmash-activemq-operator-catalog:v1.0.11 | | |
| ** ACTIVEMQ_OPERATOR_PACKAGE_MANIFEST | intersmash.activemq.operators.package_manifest | activemq-artemis-operator | amq-broker-rhel8 | |
| BOOTABLE_JAR_IMAGE_URL | intersmash.bootable.jar.image | | | |
| COMMUNITY_ACTIVEMQ_OPERATOR_PACKAGE_MANIFEST | artemis | | | |
| COMMUNITY_HYPERFOIL_OPERATOR_PACKAGE_MANIFEST | hyperfoil-bundle | | | |
| COMMUNITY_INFINISPAN_OPERATOR_PACKAGE_MANIFEST | infinispan | | | |
| COMMUNITY_KAFKA_OPERATOR_PACKAGE_MANIFEST | strimzi-kafka-operator | | | |
| COMMUNITY_KEYCLOAK_OPERATOR_PACKAGE_MANIFEST | keycloak-operator | | | |
| COMMUNITY_OPERATOR_CATALOG_SOURCE_NAME | community-operators | | | |
| COMMUNITY_WILDFLY_OPERATOR_PACKAGE_MANIFEST | wildfly | | | |
| DEFAULT_OPERATOR_CATALOG_SOURCE_NAMESPACE | openshift-marketplace | | | |
| DEPLOYMENTS_REPOSITORY_REF | intersmash.deployments.repository.ref | | | |
| DEPLOYMENTS_REPOSITORY_URL | intersmash.deployments.repository.url | | | |
| ** HYPERFOIL_OPERATOR_CATALOG_SOURCE_NAME | intersmash.hyperfoil.operators.catalog_source | community-operators | community-operators | |
| HYPERFOIL_OPERATOR_CHANNEL | intersmash.hyperfoil.operators.channel | | | |
| HYPERFOIL_OPERATOR_INDEX_IMAGE | intersmash.hyperfoil.operators.index_image | | | |
| * HYPERFOIL_OPERATOR_PACKAGE_MANIFEST | intersmash.hyperfoil.operators.package_manifest | hyperfoil-bundle | yperfoil-bundle | |
| * INFINISPAN_IMAGE_URL | intersmash.infinispan.image | | registry.redhat.io/jboss-datagrid-7/datagrid73-openshift:latest | | |
| * INFINISPAN_OPERATOR_CATALOG_SOURCE_NAME | intersmash.infinispan.operators.catalog_source | | redhat-operators | |
| INFINISPAN_OPERATOR_CHANNEL | intersmash.infinispan.operators.channel | | | |
| INFINISPAN_OPERATOR_INDEX_IMAGE | intersmash.infinispan.operators.index_image | | | |
| * INFINISPAN_OPERATOR_PACKAGE_MANIFEST | intersmash.infinispan.operators.package_manifest | | datagrid | |
| * KAFKA_OPERATOR_CATALOG_SOURCE_NAME | intersmash.kafka.operators.catalog_source | | redhat-operators | |
| ** KAFKA_OPERATOR_CHANNEL | intersmash.kafka.operators.channel strimzi-0.29.x | amq-streams-2.3.x | |
| KAFKA_OPERATOR_INDEX_IMAGE | intersmash.kafka.operators.index_image | | | |
| * KAFKA_OPERATOR_PACKAGE_MANIFEST | intersmash.kafka.operators.package_manifest | | amq-streams | |
| * KEYCLOAK_IMAGE_URL | intersmash.keycloak.image | | registry.redhat.io/rh-sso-7/sso76-openshift-rhel8:latest | |
| * KEYCLOAK_OPERATOR_CATALOG_SOURCE_NAME | intersmash.keycloak.operators.catalog_source | redhat-operators
| KEYCLOAK_OPERATOR_CHANNEL | intersmash.keycloak.operators.channel | | | |
| KEYCLOAK_OPERATOR_INDEX_IMAGE | intersmash.keycloak.operators.index_image | | | |
| * KEYCLOAK_OPERATOR_PACKAGE_MANIFEST | intersmash.keycloak.operators.package_manifest | | rhsso-operator | |
| *** KEYCLOAK_REALM_IMPORT_IMAGE_URL | intersmash.keycloak.realm_import.image | quay.io/keycloak/keycloak:21.1.1 | | |
| *** KEYCLOAK_REALM_IMPORT_OPERATOR_CATALOG_SOURCE_NAME | intersmash.keycloak.realm_import.operators.catalog_source | community-operators | | |
| *** KEYCLOAK_REALM_IMPORT_OPERATOR_CHANNEL | intersmash.keycloak.realm_import.operators.channel |fast | | |
| *** KEYCLOAK_REALM_IMPORT_OPERATOR_INDEX_IMAGE | intersmash.keycloak.realm_import.operators.index_image | registry.redhat.io/redhat/community-operator-index:v4.12 | | |
| KEYCLOAK_REALM_IMPORT_OPERATOR_PACKAGE_MANIFEST | intersmash.keycloak.realm_import.operators.package_manifest | | | |
| MAVEN_MIRROR_URL | maven-mirror.url | | | |
| MYSQL_IMAGE_URL | intersmash.mysql.image | | | |
| PGSQL_IMAGE_URL | intersmash.postgresql.image | | | |
| PRODUCT_ACTIVEMQ_OPERATOR_PACKAGE_MANIFEST | amq-broker-rhel8 | | | |
| PRODUCT_INFINISPAN_OPERATOR_PACKAGE_MANIFEST | datagrid | | | |
| PRODUCT_KAFKA_OPERATOR_PACKAGE_MANIFEST | amq-streams | | | |
| PRODUCT_KEYCLOAK_OPERATOR_PACKAGE_MANIFEST | rhsso-operator | | | |
| PRODUCT_WILDFLY_OPERATOR_PACKAGE_MANIFEST | eap | | | |
| SCRIPT_DEBUG | intersmash.openshift.script.debug | | | |
| SKIP_DEPLOY | intersmash.skip.deploy | | | |
| SKIP_UNDEPLOY | intersmash.skip.undeploy | | | |
| WILDFLY_BOMS_EE_SERVER_VERSION | bom.wildfly-ee.version | | | |
| WILDFLY_CLOUD_FEATURE_PACK_LOCATION | wildfly.cloud-feature-pack.location | | | |
| WILDFLY_EE_CHANNEL_LOCATION | wildfly.ee-channel.location | | | |
| WILDFLY_EE_FEATURE_PACK_LOCATION | wildfly.ee-feature-pack.location | | | |
| ** WILDFLY_HELM_CHARTS_BRANCH | intersmash.wildfly.helm.charts.branch | wildfly-2.3.2 | eap8-dev | |
| ** WILDFLY_HELM_CHARTS_NAME | intersmash.wildfly.helm.charts.name | wildfly | eap8 | |
| ** WILDFLY_HELM_CHARTS_REPO | intersmash.wildfly.helm.charts.repo | https://github.com/wildfly/wildfly-charts.git | https://github.com/jbossas/eap-charts.git | |
| ** WILDFLY_IMAGE_URL | intersmash.wildfly.image | quay.io/wildfly/wildfly-s2i-jdk17:latest | registry.redhat.io/jboss-eap-8-tech-preview/ | eap8-openjdk17-builder-openshift-rhel8:1.0.0.Beta | |
| WILDFLY_MAVEN_PLUGIN_ARTIFACTID | wildfly-maven-plugin.artifactId | | | |
| WILDFLY_MAVEN_PLUGIN_GROUPID | wildfly-maven-plugin.groupId | | | |
| WILDFLY_MAVEN_PLUGIN_VERSION | wildfly-maven-plugin.version | | | |
| ** WILDFLY_OPERATOR_CATALOG_SOURCE_NAME | intersmash.wildfly.operators.catalog_source | community-operators-wildfly-operator | redhat-operators | |
| ** WILDFLY_OPERATOR_CHANNEL | intersmash.wildfly.operators.channel | alpha | stable | |
| *** WILDFLY_OPERATOR_INDEX_IMAGE | intersmash.wildfly.operators.index_image | quay.io/operatorhubio/catalog:latest | | |
| ** WILDFLY_OPERATOR_PACKAGE_MANIFEST | intersmash.wildfly.operators.package_manifest | wildfly | eap | |
| ** WILDFLY_RUNTIME_IMAGE_URL | intersmash.wildfly.runtime.image | quay.io/wildfly/wildfly-runtime-jdk17:latest | registry.redhat.io/jboss-eap-8-tech-preview/eap8-openjdk17-runtime-openshift-rhel8:1.0.0.Beta | |

**Comment by tommaso-borgato** [comment]
https://docs.google.com/spreadsheets/d/1fCQ7g_sqNramx-lO0bFv6jbnUwOPxZxhHFhe_FfLuqY/edit?usp=sharing

File: deployments/intersmash-deployments-shared/intersmash-deployments-shared-wildfly/pom.xml [hunk 1]
```diff
@@ -141,7 +141,31 @@
 
     <profiles>
         <profile>
-            <id>deployments.eap</id>
+            <id>wildfly-deployments-build.eap</id>
+            <properties>
+                <!-- WildFly/EAP 8 Maven Plugin coordinates -->
+                <wildfly-maven-plugin.groupId>org.jboss.eap.plugins</wildfly-maven-plugin.groupId>
+                <wildfly-maven-plugin.artifactId>eap-maven-plugin</wildfly-maven-plugin.artifactId>
+                <wildfly-maven-plugin.version>1.0.0.Beta6-redhat-00001</wildfly-maven-plugin.version>
```
Comment (tommaso-borgato) [inline deployments/intersmash-deployments-shared/intersmash-deployments-shared-wildfly/pom.xml:149]: @fabiobrz JFD has used version `1.0.0.Final-redhat-00004` for the SAML feature pack documentation, even though it's not yet available in maven central

File: global-test.properties [hunk 2]
```diff
@@ -12,38 +12,7 @@ xtf.junit.prebuilder.synchronized=true
...
-# NOTE: we are using http://quay.io/jbossqe-eap for hosting our custom operator-catalog and operator-bundle images
-#       see https://github.com/Intersmash/intersmash/issues/32 for context
-intersmash.activemq.operators.catalog_source=intersmash-activemq-operator-index
-intersmash.activemq.operators.index_image=quay.io/jbossqe-eap/intersmash-activemq-operator-catalog:v1.0.11
-intersmash.activemq.operators.package_manifest=activemq-artemis-operator
-intersmash.activemq.operators.channel=upstream
-
-# Kafka operator settings
-intersmash.kafka.operators.channel=stable
-
```
Comment (tommaso-borgato) [inline global-test.properties:40]: @fabiobrz for the sake of users easiness of getting started quickly, why don't we leave upstream defaults here or in `test.properties`? I was asking because, by default, we have `<intersmash.test.execution.profile>community</intersmash.test.execution.profile>`

**Review by tommaso-borgato — CHANGES_REQUESTED** [review]

File: deployments/intersmash-deployments-provider/src/main/java/org/jboss/intersmash/deployments/IntersmashSharedDeploymentsProperties.java [hunk 3]
```diff
@@ -27,7 +27,6 @@ public class IntersmashSharedDeploymentsProperties {
 	private static String GROUPD_ID = "groupID";
 	private static String DEPLOYMENTS_PROVIDER_PATH = "deploymentsProviderPath";
 	private static String WILDFLY_DEPLOYMENTS_BUILD_PROFILE = "wildflyDeploymentsBuildProfile";
```
[the remark this answers is not in the document]
Comment (fabiobrz) [reply deployments/intersmash-deployments-provider/src/main/java/org/jboss/intersmash/deployments/IntersmashSharedDeploymentsProperties.java:29]: Yep, exactly. I've followed a pattern we've seen in other projects.

File: deployments/intersmash-deployments-shared/intersmash-deployments-shared-wildfly/pom.xml (hunk 1 above)
Comment (fabiobrz) [reply deployments/intersmash-deployments-shared/intersmash-deployments-shared-wildfly/pom.xml:149]: I chose to keep everything EAP 8 related to the latest released version, i.e. EAP 8 Beta.
This way we can be sure the CI checks that are run in the OpenShift CI infra will find the required bits.
So, we'll update everything to align with EAP 8 when it will be GA.

File: global-test.properties (hunk 2 above)
Comment (fabiobrz) [reply global-test.properties:40]: You're right, this makes sense. I'll fix it, thanks!

File: testsuite/src/main/java/org/jboss/intersmash/testsuite/junit5/categories/NotForProductizedExecutionProfile.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import org.junit.jupiter.api.Tag;
+
+/**
+ * Mark test that tests AMQ Streams service.
+ * Used per class.
+ */
+@Tag("ts.not-for-prod")
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ java.lang.annotation.ElementType.TYPE })
+public @interface NotForProductizedExecutionProfile {
```
[the remark this answers is not in the document]
Comment (fabiobrz) [reply testsuite/src/main/java/org/jboss/intersmash/testsuite/junit5/categories/NotForProductizedExecutionProfile.java:16]: Yep, exactly.

File: testsuite/src/test/java/org/jboss/intersmash/testsuite/provision/openshift/WildflyOperatorProvisionerTest.java [hunk 5]
```diff
@@ -41,11 +42,22 @@ public class WildflyOperatorProvisionerTest {
 
 	private static WildflyOperatorProvisioner initializeOperatorProvisioner() {
 		WildflyOperatorProvisioner operatorProvisioner = new WildflyOperatorProvisioner(
+
 				new WildflyOperatorApplication() {
 					@Override
 					public WildFlyServer getWildflyServer() {
+						String image;
+						if (IntersmashTestsuiteProperties.isCommunityTestExecutionProfileEnabled()) {
+							image = "quay.io/wildfly-quickstarts/wildfly-operator-quickstart:latest";
```
[the remark this answers is not in the document]
Comment (fabiobrz) [reply testsuite/src/test/java/org/jboss/intersmash/testsuite/provision/openshift/WildflyOperatorProvisionerTest.java:51]: I would not pollute the Intersmash config API with this property, which is mostly related to a specific testsuite test.

**Comment by fabiobrz** [comment]
The tests that use the prod deliverables will be run periodically once https://github.com/openshift/release/pull/43119 will be merged.

**Comment by tommaso-borgato** [comment]
LGTM, merging

**Review by tommaso-borgato — APPROVED** [review]
